### PR TITLE
[22.05] Mount ``$TMPDIR``, ``$TMP`` and ``$TEMP`` into container

### PR DESCRIPTION
**Pull request by mvdbeek**
## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by mvdbeek** [comment]
@nuwang how do you guys handle tools that create files in `$TMP` ? This seems to have been broken for docker, I've fixed this here, but it's unclear how I would fix this for kubernetes ? Have you configured this somehow in the helm chart ?

**Comment by mvdbeek** [comment]
It also appears that `<env id="TMPDIR">/tmp</env>` breaks the job in a weird way locally ...

**Comment by mvdbeek** [comment]
I don't think this breaks anything that wasn't previously broken in kubernetes, and the test errors are unrelated and fixed on dev, if anyone wants to merge it.

File: lib/galaxy/tool_util/deps/container_classes.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def add_var(name, value):
                 # and Galaxy.
                 defaults += ",$tmp_directory:/tmp:rw"
             else:
-                defaults += ",$_GALAXY_JOB_TMP_DIR:rw"
+                defaults += ",$_GALAXY_JOB_TMP_DIR:rw,$TMPDIR:rw,$TMP:rw,$TEMP:rw"
```
Comment (bernt-matthias) [inline lib/galaxy/tool_util/deps/container_classes.py:238]: Do we need to check if these variables exist / are defined?

Comment (mvdbeek) [reply lib/galaxy/tool_util/deps/container_classes.py:238]: We can't (at least without some additional shell hacking), but if they don't exist nothing happens, try `docker run -it -v "::rw" busybox sh`

Comment (bernt-matthias) [reply lib/galaxy/tool_util/deps/container_classes.py:238]: Cool 😎

**Review by bernt-matthias — APPROVED** [review]
Tested this successfully (with planemo) on one of the tests failing here 

https://github.com/galaxyproteomics/tools-galaxyp/actions/runs/3549279419 (for this PR https://github.com/galaxyproteomics/tools-galaxyp/pull/653)

Edit: most of the failing tests there have the same symptom. Something like `Could not remove directory 20221125_171454_dee36a6fa1f6_15_1!`  in stderr.

**Review by jmchilton — APPROVED** [review]

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**